Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -799,256 +771,202 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         toolsVersion: ToolsVersion,
         observabilityScope: ObservabilityScope,
         delegate: Delegate?,
-        delegateQueue: DispatchQueue?,
-        callbackQueue: DispatchQueue,
-        completion: @escaping (Result<EvaluationResult, Error>) -> Void
-    ) throws {
+        delegateQueue: DispatchQueue?
+    ) async throws -> EvaluationResult {
         // The compiler has special meaning for files with extensions like .ll, .bc etc.
         // Assert that we only try to load files with extension .swift to avoid unexpected loading behavior.
         guard manifestPath.extension == "swift" else {
-            return callbackQueue.async {
-                completion(.failure(InternalError("Manifest files must contain .swift suffix in their name, given: \(manifestPath).")))
-            }
+            throw InternalError("Manifest files must contain .swift suffix in their name, given: \(manifestPath).")
         }
 
-        var evaluationResult = EvaluationResult()
+        return try await self.tokenBucket.withToken {
```

Comment:
I don't have the background, but it is unclear why this diff is introducing the withToken indentation

Reply:
`withToken` introduces indentation much earlier, since with stricter concurrency checks `withToken` has to initialize state like `evaluationResult` much earlier in its scope than was previously possible with unchecked Dispatch closures. Previously this state was captured with no sendability checks whatsoever. Much broader `withToken` scope allows us to initialize all state in the same closure, fixing sendability warnings and errors.